File: MessageApp/Models/Category.cs


using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MessageApp.Models
{
    public class Category
    {
        [Key]
        public int CategoryID { get; set; }
        public string? Title { get; set; }
        public DateTime Date { get; set; }
        public ICollection<Topic> Topics { get; set; }
    }
}

File: MessageApp/client/src/app/_models/Category.ts
import { ICategory } from "../interfaces/ICategory";

export class Category implements ICategory {
  public categoryID: number;
  public title: string;
  public date: string;
  public time: string;
  constructor(categoryID: number = -1, title: string = '', date: string = '', time: string = '') {
    this.categoryID = categoryID;
    this.title = title;
    this.date = date;
    this.time = time;
  }

}
